valohai_yaml/pipelines/conversion.py

```diff
@@ -55,22 +54,25 @@ def convert_deployment_node(self, node: DeploymentNode) -> ConvertedObject:
     def convert_executionlike_node(self, node: Union[ExecutionNode, TaskNode]) -> ConvertedObject:
         node_data = node.serialize()
         step_name = node_data.pop("step")
-        override = node_data.pop("override", {})
         step = self.config.get_step_by(name=step_name)
         if not step:  # pragma: no cover
             raise ValueError(f"Step {step_name} not found in {self.config}")
+
         step_data = step.serialize()
-        step_data.update(override)
+
+        inputs_from_node = node.get_input_defaults()
+        inputs_from_step = step.get_input_defaults()
+        step_data["inputs"] = {**inputs_from_step, **inputs_from_node}
 
         parameters_from_node = node.get_parameter_defaults()
         parameters_from_step = step.get_parameter_defaults(include_flags=True)
-        step_data["parameters"] = parameters_from_step
-        step_data["parameters"].update(parameters_from_node)
+        step_data["parameters"] = {**parameters_from_step, **parameters_from_node}
```

Comment:
Hmm... It looks like `get_*_defaults()` doesn't return inputs or parameters for which no defaults have been set? Additionally, the value returned from the dict is just a list of defaults, not the input or parameter definition (as I suppose we expect..?)

I may be missing something here though. 😅

Reply:
I will check it

Reply:
well, I though that idea is to override values so if value in override but not in step, it added it into node. So imho it should keep working that way, but I also can miss smth here. @tokkoro do I?
I updated tests for covering this use case.

Reply:
Isn't this the place where we generate the template for nodes? the template should include only name-value pairs in dictionary for parameters and not the definition. Roi will find them in the commit that is defined in node and then add these values on those when creating the actual Execution.

Reply:
So to summarize, you two are talking about two different things here. @akx I think you are thinking of a different place, as this is the template for execution/task node and shouldn't include parameter definitions (those are found by commit+step_name of the node). @euvolen-valohai you are correct about the override logic, that if something has default just in node override and not in the step it should have the overriding default value here after.